src/snowflake/cli/api/sql_contract.py

```diff
@@ -0,0 +1,134 @@
+from contextlib import contextmanager
+
+from snowflake.cli.api.constants import ObjectType
+from snowflake.cli.api.entities.common import get_sql_executor
+from snowflake.cli.api.entities.utils import generic_sql_error_handler
+from snowflake.cli.api.errno import DOES_NOT_EXIST_OR_CANNOT_BE_PERFORMED
+from snowflake.cli.api.exceptions import CouldNotUseObjectError
+from snowflake.connector import ProgrammingError
+
+
+class UnknownSQLError(Exception):
+    def __init__(self, msg):
+        super().__init__(f"Unknown SQL error occurred. {msg}")
+
+
+class CannotUseRoleError(CouldNotUseObjectError):
+    def __init__(self, role):
+        self.role = role
+        super().__init__(ObjectType.ROLE, role)
+
+
+class SQLService:
```

Comment:
```suggestion
class SqlService:
```
to match our other `Sql...` classes